cosmwasm/contract.ts

```diff
@@ -0,0 +1,433 @@
+        options.description = options.description || `${operation} ${chainsList} on ITS Hub`;
+    }
+
+    return executeContractMessage(client, config, options, 'InterchainTokenService', msg, fee);
+};
+
+const registerProtocol = async (
+    client: ClientManager,
+    config: ConfigManager,
+    options: ContractCommandOptions,
+    _args?: string[],
+    fee?: string | StdFee,
+): Promise<void> => {
+    const serviceRegistry = config.validateRequired(config.getContractConfig('ServiceRegistry').address, 'ServiceRegistry.address');
+    const router = config.validateRequired(config.getContractConfig('Router').address, 'Router.address');
+    const multisig = config.validateRequired(config.getContractConfig('Multisig').address, 'Multisig.address');
+
+    const msg = [
+        JSON.stringify({
+            register_protocol: {
+                service_registry_address: serviceRegistry,
+                router_address: router,
+                multisig_address: multisig,
+            },
+        }),
+    ];
+
+    if (!options.title || !options.description) {
+        options.title = options.title || 'Register Protocol contracts on Coordinator';
+        options.description = options.description || 'Register Protocol contracts on Coordinator';
+    }
+
+    return executeContractMessage(client, config, options, 'Coordinator', msg, fee);
+};
+
+const registerDeployment = async (
+    client: ClientManager,
+    config: ConfigManager,
+    options: ContractCommandOptions,
+    _args?: string[],
+    fee?: string | StdFee,
+): Promise<void> => {
+    const { chainName } = options;
+    const coordinator = new CoordinatorManager(config);
+    const message = coordinator.constructRegisterDeploymentMessage(chainName);
+    const msg = [JSON.stringify(message)];
+
+    if (!options.title || !options.description) {
+        options.title = options.title || `Register ${chainName} deployment on Coordinator`;
+        options.description = options.description || `Register ${chainName} deployment on Coordinator`;
+    }
+
+    return executeContractMessage(client, config, options, 'Coordinator', msg, fee);
+};
+
+const createRewardPools = async (
+    client: ClientManager,
+    config: ConfigManager,
+    options: ContractCommandOptions,
+    _args?: string[],
+    fee?: string | StdFee,
+): Promise<void> => {
+    const { chainName, epochDuration, participationThreshold, rewardsPerEpoch } = options;
+
+    if (!participationThreshold) {
+        throw new Error('Participation threshold is required');
+    }
+    if (!epochDuration) {
+        throw new Error('Epoch duration is required');
+    }
+    if (!rewardsPerEpoch) {
+        throw new Error('Rewards per epoch is required');
+    }
+
+    let parsedThreshold: string[];
+    try {
+        parsedThreshold = JSON.parse(participationThreshold);
+        if (!Array.isArray(parsedThreshold)) {
+            throw new Error('Participation threshold must be a JSON array');
```

Comment:
create a precise validation method for a threshold value and then use validateParameters